### PR TITLE
Autoload HuggingFace model/tokenizer

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
This PR adds support for loading a huggingface model and tokenizer from the information saved in the the composer checkpoint that is added by #1744.

# What issue(s) does this change relate to?
Design here: https://www.notion.so/Autoload-Huggingface-models-for-finetuning-59dda720b7ae4bee90d3594a5ab731d9
Closes: [CO-1109](https://mosaicml.atlassian.net/browse/CO-1109) and [CO-1110](https://mosaicml.atlassian.net/browse/CO-1110)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: composer/models/huggingface.py
Comment (mvpatel2000) [inline composer/models/huggingface.py:39]: should be note syntax? `.. note::`, see Trainer for reference

File: composer/models/huggingface.py
Comment (mvpatel2000) [inline composer/models/huggingface.py:159]: wrap in MissingConditionalImportError?

File: docs/source/doctest_fixtures.py
Comment (mvpatel2000) [inline docs/source/doctest_fixtures.py:88]: unused line?

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Would want a review from NLP person as well though before merging?

File: composer/models/huggingface.py
Comment (dakinggg) [reply composer/models/huggingface.py:159]: It seemed like we only did that in class inits, not everywhere the import is used. But i guess you can use this function without calling the class init? I can wrap this if you think that makes sense.

File: docs/source/doctest_fixtures.py
Comment (dakinggg) [reply docs/source/doctest_fixtures.py:88]: Not unused, need to import here so it is accessible in doctests

File: composer/models/huggingface.py
Comment (dakinggg) [reply composer/models/huggingface.py:39]: ah thanks will do

File: composer/models/huggingface.py
Comment (dakinggg) [reply composer/models/huggingface.py:160]: wrapped

**Comment by dakinggg** [comment]
@alextrott16 @A-Jacobson This is ready for review now

File: composer/models/huggingface.py
Comment (alextrott16) [inline composer/models/huggingface.py:103]: The naming `model_init_kwargs` strikes me as a bit vague. I think we tend to use `model_config` as the name for the config default overrides. In any case, I suggest naming it something that mentions the config, since that is what it directly changes.

File: composer/models/huggingface.py
Comment (alextrott16) [inline composer/models/huggingface.py:199]: Since we live and die by the YAML, is there a way to support string arguments for `model_instantiation_class` (i.e. if the input is a string, use that to import the class and then use the imported class)? Maybe this is already solved with Hydra, but we don't want to force that.

**Review by alextrott16 — APPROVED** [review]
One name suggestion and one feature request. I leave them to you to decide whether they are worth doing. So, LGTM!
Disclaimer: if there are issues with the tests, it's very possible I have overlooked them, but they seem OK at a glance.

File: composer/models/huggingface.py
Comment (dakinggg) [reply composer/models/huggingface.py]: :+1: changed to `model_config_kwargs`

File: composer/models/huggingface.py
Comment (dakinggg) [reply composer/models/huggingface.py:199]: yeah, should be pretty easy to add, will do.

**Pull request merged**
